securedrop/sdconfig.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+import config as _config
+
+import typing
+# https://www.python.org/dev/peps/pep-0484/#runtime-or-type-checking
+if typing.TYPE_CHECKING:
+    # flake8 can not understand type annotation yet.
+    # That is why all type annotation realted import
+    # statements has to be marked as noqa.
+    # http://flake8.pycqa.org/en/latest/user/error-codes.html?highlight=f401
+    from typing import List, Dict  # noqa: F401
+
+
+class SDConfig(object):
+    def __init__(self):
+        # type: () -> None
+        self.FlaskConfig = _config.FlaskConfig
+        self.JournalistInterfaceFlaskConfig = \
+            _config.JournalistInterfaceFlaskConfig
+        self.SourceInterfaceFlaskConfig = _config.SourceInterfaceFlaskConfig
+
+        if _config.DATABASE_ENGINE != "sqlite":
+            self.DATABASE_USERNAME = _config.DATABASE_USERNAME  # type: ignore
+            self.DATABASE_PASSWORD = \
+                _config.DATABASE_PASSWORD   # type: ignore
+            self.DATABASE_HOST = _config.DATABASE_HOST  # type: ignore
+            self.DATABASE_NAME = _config.DATABASE_NAME  # type: ignore
+        else:
+            self.DATABASE_FILE = _config.DATABASE_FILE
+
+        if hasattr(_config, 'CUSTOM_HEADER_IMAGE'):
+            self.CUSTOM_HEADER_IMAGE = \
+                _config.CUSTOM_HEADER_IMAGE  # type: ignore
+
+        if hasattr(_config, 'SUPPORTED_LOCALES'):
```

Comment:
In Python 2 `hasattr` will return false on any exception - not just an `AttributeError` as one might expect, so an alternative here that does not swallow exceptions (since this can create debugging issues in the future) would be to adopt an [EAFP](https://blogs.msdn.microsoft.com/pythonengineering/2016/06/29/idiomatic-python-eafp-versus-lbyl/) approach to handle the `AttributeError` only (as shown [in this blog](https://hynek.me/articles/hasattr/) - kudos to @kushaldas for pointing me to this). What do you think @dachary @heartsucker?

Reply:
> Which is almost never what you want because it shadows errors in properties:

Although this does not apply here given the simplistic nature of config.py as we have it now, I must confess I did not know at all about this. And I'm happy to have been educated, thanks @kushaldas 

I'm 100% ok to replace all hasattr in this function with try / except even though it is longer